gw_challenge/ingested_program/ingestion.py

```diff
@@ -8,7 +8,7 @@
 
 
 # expected version pattern for requirements
-VERSION_PATTERN = re.compile("^[0-9].[0-9].[0-9]$")
+VERSION_PATTERN = re.compile("^[N!]N(.N)*[{a|b|rc}N][.postN][.devN]$")
```

Comment:
sorry I only saw it after merging, but upon testing, this version pattern doesn't seem to be doing what we want? For example, version "2.15.0" does not match the pattern, is it expected behaviour? Thanks 

@egrace479 @work4cs

Reply:
Yeah, that does not seem to be working. @work4cs, this was your update on ours, could you please resolve this or revert both back to the pattern I initially set up? Thanks!

Reply:
I agree, that seems like the right thing to use rather than come up with our custom reinvention. Nice find @work4cs.

Reply:
@work4cs interesting solution! Could you please add it to the code and make another pull request? Thanks!

Reply:
thanks for pointing out to the changes, I will transfer them to this repo!

Reply:
Sounds like a good plan @work4cs, thanks!